src/components/angular2-modal/models/dialog-ref.ts

```diff
@@ -1,20 +1,39 @@
 import { ComponentRef } from '@angular/core';
-import { PromiseWrapper } from '@angular/core/src/facade/promise';
+import { PromiseWrapper, PromiseCompleter } from '@angular/core/src/facade/promise';
 
 /**
  * API to an open modal window.
  */
 export class DialogRef<T> {
-    contentRef: ComponentRef<any>;
+    /** 
+     * The reference to the component ref.
+     * This can be return undefined, since the componentRef is set only when the modal is shown.
+     * Use componentRefPromise to get a promise that will resolve when the componentRet is set.
+     * @return {ComponentRef<any>}
+     */
+    get contentRef(): ComponentRef<any> { return this._contentRef; }
+    set contentRef(value: ComponentRef<any>) {
+        this._contentRef = value;
+        this._conponentRefDeferred.resolve(value);
+    }
+
+    /**
+     * A promise that is resolved when the component ref is set.
+     * @return {Promise<ComponentRef<any>>}
+     */
+    get contentRefPromise() { return this._conponentRefDeferred.promise; }
```

Comment:
>What is the use case for having access to the contentRef? it's internal.
To access the instance of the Component.
But I agree that it does not happens often.

>For example, a PR for beforeDismiss and beforeClose and a different one for the other change.

I have to agree with you here now. I'll remove the contentRef changes so the other part that is more useful can be merged